timetracker/vms/test/models/test_time_record_model.py

```diff
@@ -93,3 +93,42 @@ def test_3_time_string_conversion(time_record_factory):
         f'{t.time_end:%I:%M %p} on {t.time_start:%m/%d/%Y}.'
     )
     assert str(t) == expected
+
+
+def test_total_time_no_end_time(time_record_factory):
+    """
+    Case with no end time
+    """
+    d = datetime.datetime(2018, 10, 1, 15, 26)
+    t = time_record_factory(time_start=d, time_end=None)
+    expected = None
+    assert t.total_time == expected
+
+
+def test_total_time_with_start_and_end_times(time_record_factory):
+    """
+    Case with start and end times. Same day, 9-5 workday
+    """
+    ts = datetime.datetime(2018, 10, 1, 9, 0)
+    te = datetime.datetime(2018, 10, 1, 17, 0)
+
+    t = time_record_factory(time_start=ts, time_end=te)
+
+    expected = float(8.0)
+
+    assert t.total_time == expected
+
+
+def test_total_time_rounding_start_and_end_time(time_record_factory):
+    """
+    Case with start and end times. Same day, 9-5 workday
+    """
+    ts = datetime.datetime(2018, 10, 1, 9, 0)
+    te = datetime.datetime(2018, 10, 1, 17, 59)
+
+    t = time_record_factory(time_start=ts, time_end=te)
+    # round to 17.5 - 9 = 8.5
+    expected = float(8.5)
+    print(t.total_time)
+
+    assert t.total_time == expected
```

Comment:
Comparing floats can often go wonky. I would recommend using [`pytest.approx`](https://docs.pytest.org/en/latest/reference.html#pytest-approx) to compare floats.

```suggestion
    assert t.total_time == pytest.approx(expected)
```